content/en/docs/tasks/administer-cluster/nodelocaldns.md

```diff
@@ -2,61 +2,67 @@
+## Motivation
 
+Without a local DNS cache on each Node, Pods in ClusterFirst DNS mode access the `kube-dns` application via its serviceIP in order to resolve DNS queries. Your cluster translates that serviceIP to a kube-dns or CoreDNS endpoint via kube-proxy.
+  With this add-on, Pods send DNS queries to a caching agent running on the same node. The local caching agent queries your cluster's kube-dns application when resolving cluster DNS queries. For cache hits, the local agent serves the response directly.
 
-## Motivation
+In detail:
 
-* With the current DNS architecture, it is possible that Pods with the highest DNS QPS have to reach out to a different node, if there is no local kube-dns/CoreDNS instance.  
-Having a local cache will help improve the latency in such scenarios.
+* With the current DNS architecture, it is possible that Pods with the
+  highest DNS QPS have to reach out to a different node, if there is no
+  local kube-dns / CoreDNS instance.
+  Having a local cache help improve the latency in such scenarios.
 
-* Skipping iptables DNAT and connection tracking will help reduce [conntrack races](https://github.com/kubernetes/kubernetes/issues/56903) and avoid UDP DNS entries filling up conntrack table.
+* Skipping iptables DNAT and connection tracking helps to reduce [conntrack races](https://github.com/kubernetes/kubernetes/issues/56903) and to avoid UDP DNS entries filling up conntrack table.
 
-* Connections from local caching agent to kube-dns servie can be upgraded to TCP. TCP conntrack entries will be removed on connection close in contrast with UDP entries that have to timeout ([default](https://www.kernel.org/doc/Documentation/networking/nf_conntrack-sysctl.txt) `nf_conntrack_udp_timeout` is 30 seconds)
+* Connections from local caching agent to kube-dns servie can be upgraded to TCP. TCP conntrack entries will be removed on connection close in contrast with UDP entries that have to time out ([default](https://www.kernel.org/doc/Documentation/networking/nf_conntrack-sysctl.txt) `nf_conntrack_udp_timeout` is 30 seconds)
 
-* Upgrading DNS queries from UDP to TCP would reduce tail latency attributed to dropped UDP packets and DNS timeouts usually up to 30s (3 retries + 10s timeout). Since the nodelocal cache listens for UDP DNS queries, applications don't need to be changed.
+* Upgrading DNS queries from UDP to TCP would reduce tail latency attributed to dropped UDP packets and DNS timeouts usually up to 30s (3 retries + 10s timeout). Since the node-local cache container listens for UDP DNS queries, you don't need to change applications.
 
 * Metrics & visibility into dns requests at a node level.
 
-* Negative caching can be re-enabled, thereby reducing number of queries to kube-dns service.
-
-## Architecture Diagram
+* Negative caching can be re-enabled, thereby reducing number of queries to the kube-dns service.
 
-This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
```

Comment:
As this is a Task page I'm tempted to omit these details, or to move them to the end of the page. What do people think?